core/src/main/java/com/linecorp/armeria/server/http/MaxConnectionHandler.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+
+@Sharable
+public final class MaxConnectionHandler extends ChannelInboundHandlerAdapter {
```

Comment:
How about:

- Moving this class to `com.linecorp.armeria.internal`? (Note that this can be used for non-HTTP scenarios as well.)
- Renaming this class to `ConnectionLimitingHandler`?